app/src/main/java/com/m/helper/MobileService.java

```diff
@@ -0,0 +1,159 @@
+package com.m.helper;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.m.helper.configModel.Keycloak;
+import com.m.helper.configModel.MobileServiceJSON;
+import com.m.helper.configModel.Push;
+import com.m.helper.configModel.Service;
+import com.m.helper.configModel.SyncApp;
+import com.m.helper.configModel.config.AndroidConfig;
+
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+public class MobileService implements IMobileService {
+
+    private static MobileService INSTANCE = null;
+    private MobileServiceJSON config = new MobileServiceJSON();
+    private AssetManager assetManager;
+    private Gson gson = new Gson();
+
+    private final String SYNC_APP = "sync-app";
+    private final String KEYCLOAK = "keycloak";
+    private final String PUSH = "push";
+
+    private final String FILE_LOCATION = "config/mobile-services.json";
+
+    private MobileService(Context context){
+        System.out.println("APP : Setting up Mobile Services");
+        assetManager = context.getAssets();
+        readMobileServiceJSON();
+    }
+
+    public static MobileService getInstance(Context context) {
+        if (INSTANCE == null) {
+            INSTANCE = new MobileService(context);
+        }
+        return INSTANCE;
+    }
+
+    public static MobileService getInstance() {
+        return INSTANCE;
+    }
```

Comment:
In order for this method to work, user will need to call `getInstance(Context context)` first. However, there is no docs or comments to explain about that.

If we think about it, `MobileService` will always need a `Context` object to initialise itself. In this case, it is better to create a explicit `Init` method that can take in the required parameters. Then when other methods like `getInstance` is called, it can throw exceptions if the class is not inited yet.